### DISCUSSION TITLE
CLI command for Updating Machinedeployment

### DISCUSSION THREAD

**Discussion by ThWoywod** [kubermatic/kubeone#2822] [Ideas]
Hello everyone, 
I'm thinking about extending the kubeone cli and creating a way to update only certain machinde deployments.

My consideration comes from the fact that I have the use case that I need to execute program logic between the update of individual machinde deployments.

The CLI already allows to say with an "apply" that the machine deployments are not updated. Unfortunately there is no way to say ok now please update only the machine deployments. that means I have to manually adjust the customizations in the machine deployments now or I run an update of everything again afterwards.

**Comment by kron4eg** [comment]
MachineDeployments are just usual kubernetes API objects, please treat them accordingly, as in ether script somewhere `kubectl apply -f ./my-machinedeployments/` or any other mean of deploying kubernetes objects.

**Reply by ThWoywod** [reply]
Totally true, but the machinedeployments i actually defen in my terraform output file. Yes i could pull it out there but I thought this was the recommended way.
There is also no known way to generate the machinedeployment file. from the terraform output file. That's why I thought it would be a useful extension.

**Reply by xmudrii** [reply]
> Yes i could pull it out there but I thought this was the recommended way.

The way it works is that those MachineDeployments defined in Terraform output are _initial_ MachineDeployments. That said, that data is used only to create the given MachineDeployment(s) and after that it's up to the user to manage those MachineDeployments with `kubectl` or any other preferred tool.

> There is also no known way to generate the machinedeployment file. from the terraform output file.

You can use `kubeone config machinedeployments` command for that.

I'd also recommend looking into `kubectl patch` command if you only want to change version, that way you can do it non-interactive inside your script.

**Reply by ThWoywod** [reply]
Awesome, this is exactly what I need and would be enough for me.

**Reply by vuducdung1308** [reply]
awesome, thank you for solution
